src/main.py

```
@@ -246,31 +289,113 @@ async def respond_with_bot_message(update: Update) -> None:
+            error("Telegram timeout while sending video. %s", e)
+        except (NetworkError, TelegramError) as e:
+            await update.message.reply_text(f"Error sending video: {str(e)}. Please try again later.")
+        finally:
+            video_file.close()
+
+    # Send the group of videos
+    elif isinstance(video, list):
+        media_group = []
+        opened_files = []
+        for video_file in video:
+            file = open(video_file, 'rb')
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)